xlml/apis/task.py

```diff
@@ -374,6 +374,9 @@ def run_model(self) -> DAGNode:
     """
     with TaskGroup(group_id="run_model") as group:
       workload_id = xpk.generate_workload_id(self.task_test_config.benchmark_id)
+      gcs_path = name_format.generate_gcs_folder_location(
```

Comment:
TaskFlow will implicitly add the dependencies, so I left it out of L400. Let me know if I should make it explicit.

For the shared GCS bucket use case, how would we configure the multiple workloads in one XPK task? Would we need a `List[task_test_config]` and `List[gcp_config]` in `XpkTask` definition?

If we wan to keep the two as distinct XpkTask with the same GCS path, we could make the GCS path an optional argument to `run`, and if it's missing then do the generation via `name_format.generate_gcs_folder_location`. So something like:

```python
def run(gcs_path: XComArg[str]):
   ...
   gcs_path = gcs_path or name_format.generate_gcs_folder_location(...)
```

The in cases where we need to link multiple workloads to the same bucket, do something like:

```python
shared_gcs_path = name_format.generate_gcs_folder_location('shared-task-id')
# Task to convert chkpt
get_gke_config(...).run(gcs_path=shared_gcs_path)
# Task to run model
get_gke_config(...).run(gcs_path=shared_gcs_path)
```

Reply:
```
TaskFlow will implicitly add the dependencies, so I left it out of L400. Let me know if I should make it explicit.
```

Let's update that as well for less confusion for readers like me (since we already have L400 there). We could specify something like (workload_id, generate_gcs_folder_location) >> run_workload >> wait_for_workload_completion

Right, the proposal makes sense. I am also thinking to try this `xcom_push` and `xcom_pull` in tasks. For instance, inside of this [method](https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/4a020882beb90c1df8b5595e55a3fb32f429bb8b/xlml/utils/name_format.py#L51),  it queries key first:
* if has value: directly return;
* if no value: generate and return.

If this works, we don't need `shared_gcs_path = name_format.generate_gcs_folder_location('shared-task-id')` and `def run(gcs_path: Optional[XComArg[str]] = None)`.

This is xcom value from `generate_gcs_folder_location` task: https://screenshot.googleplex.com/PoFqg73CUNNHgZr. It's already there.